cura/Settings/MachineManager.py

```diff
@@ -550,6 +550,18 @@ def copyValueToExtruders(self, key: str):
             if extruder_stack != self._active_container_stack and extruder_stack.getProperty(key, "value") != new_value:
                 extruder_stack.userChanges.setProperty(key, "value", new_value)  # TODO: nested property access, should be improved
 
+    ## Copy the value of all settings of the current extruder to all other extruders as well as the global container.
+    @pyqtSlot()
+    def copyAllValuesToExtruders(self):
+        for key in self._active_container_stack.userChanges.getAllKeys():
```

Comment:
I think you can simplify/optimize the function this way:
```
extruder_stacks = [all extruders which is not the currently active one]
for key in active_extruder's userchanges keys:
    for extruder_stack in extruder_stacks:
        copy values over
```
This way, you don't need to reconstruct the `extruder_stacks` list in each iteration.